RecCalorimeter/src/components/SimulateSiPMwithOpticalPhoton.h

```diff
@@ -0,0 +1,104 @@
+#ifndef RECCALORIMETER_SimulateSiPMwithOpticalPhoton_H
+#define RECCALORIMETER_SimulateSiPMwithOpticalPhoton_H
+
+// EDM4HEP includes
+#include "edm4hep/RawCalorimeterHitCollection.h"
+#include "edm4hep/CalorimeterHitCollection.h"
+#include "edm4hep/RawTimeSeriesCollection.h"
+#include "edm4hep/TimeSeriesCollection.h"
+
+// k4FWCore includes
+#include "k4FWCore/DataHandle.h"
+
+// Gaudi includes
+#include "Gaudi/Algorithm.h"
+#include "GaudiKernel/ToolHandle.h"
+#include "GaudiKernel/IRndmGenSvc.h"
+#include "GaudiKernel/RndmGenerators.h"
+
+// Check for SiPMSensor header location (similar to how DigiSiPM handles this)
+#if __has_include("SiPMSensor.h")
+#include "SiPMSensor.h"
+#elif __has_include("sipm/SiPMSensor.h")
+#include "sipm/SiPMSensor.h"
+#endif
+
+/** @class SimulateSiPMwithOpticalPhoton
+ *
+ *  Algorithm for digitizing the SiPM response
+ *  This digitizer takes RawTimeSeriesCollection and RawCalorimeterHitCollection as input
+ *  and produces TimeSeriesCollection and CalorimeterHitCollection as output.
+ *
+ *  The algorithm simulates SiPM response including effects like:
+ *  - Dark count rate (DCR)
+ *  - Crosstalk (Xt)
+ *  - Afterpulsing
+ *  - Cell recovery time
+ *  - Signal rise and fall times
+ *
+ *  @author Sanghyun Ko
+ *  @author Sungwon Kim
+ *  @date   2025-03-18
+ */
+
+class SimulateSiPMwithOpticalPhoton : public Gaudi::Algorithm {
+public:
+  SimulateSiPMwithOpticalPhoton(const std::string& name, ISvcLocator* svcLoc);
+  virtual ~SimulateSiPMwithOpticalPhoton() {};
+
+  StatusCode initialize() override;
+  StatusCode execute(const EventContext&) const override;
+  StatusCode finalize() override;
+
+private:
+  // integral function for the wavelength random generation
+  std::vector<double> integral(const std::vector<double>& wavelen, const std::vector<double>& yval) const;
+
+  // Random Number Service
+  SmartIF<IRndmGenSvc> m_randSvc;
+  Rndm::Numbers m_rndmUniform;
+
+  // Input collections
+  mutable DataHandle<edm4hep::RawCalorimeterHitCollection> m_rawHits{"DRcaloSiPMreadoutRawHit", Gaudi::DataHandle::Reader, this};
```

Comment:
For these three lines, can you make the name of the input collection configurable from the steering file? Either as you did with `SimulateSiPMwithEdep` or with something like https://github.com/HEP-FCC/k4RecCalorimeter/blob/main/RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp#L26 .